src/spikeinterface/core/baserecording.py

```diff
@@ -509,6 +509,26 @@ def reset_times(self):
             rs.t_start = None
             rs.sampling_frequency = self.sampling_frequency
 
+    def shift_start_time(self, shift, segment_index=None):
```

Comment:
Typing x D ?

The function could also be called "shift_times" to avoid making a reference to the internal detail of "t_start", etc but I think this works as well.

Reply:
I think I agree. I would name the function whatever is user facing more. So if the "times" are then times, but if the start is more important than start.

Reply:
Typing!! Need to add! 🎉 

Yes I think I agree, my only thought is, is there a world where 'shift_times' is not clear because it might imply the possibility to make completely non-uniform shifts e.g. supply a vector of shifts? The only benefit of something like `shift_start_times` is it provides a reference point from where the shift will occur. Maybe it is also more intuitive as if you have a recording it is natural to imagine starting it at a different time. But maybe it is confusing with the `t_start` implementation. I'm basically 50/50 so happy to take the way of the majority!

Reply:
I am in the same train. 50/50, feel free to pick what your hearth thinks is better!

Reply:
Haha okay ❤️ I will decide at the last moment! Thanks for the test suggestions, also good point on the preprocessing steps I am not sure either, I opened #3534